### PR TITLE
Update actions to allow for PRs from forks

### PR DESCRIPTION
**Pull request by DonoA**
Attempting to address issues with merging PRs from forked repos.

This appears to be an issue with the secret management in Github and how it interacts with forked repos.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,11 +1,30 @@
-name: Publish
+name: Release Package
 
 on:
   release:
     types: [published]
   workflow_dispatch:
```
Comment (zromano) [inline .github/workflows/release.yml]: Do we need this `workflow_dispatch` line in the release.yml and Test-pr.yml? Looks like it's empty on both

File: .github/workflows/test-pr.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Run Mock API Tests
+        env:
+          SMARTSHEET_ACCESS_TOKEN: ${{secrets.SMARTSHEET_ACCESS_TOKEN}}
```
Comment (zromano) [inline .github/workflows/test-pr.yaml:20]: PRs will not have access to any secrets, so `SMARTSHEET_ACCESS_TOKEN` will just be blank here.

Comment (DonoA) [reply .github/workflows/test-pr.yaml:20]: That's what it seems like to me too. I wanted to leave it here as the owner of the fork can provide their own secret if they would like to test against their own account

File: .github/workflows/release.yml (hunk 1 above)
Comment (DonoA) [reply .github/workflows/release.yml]: Let's remove it and see

File: .github/workflows/test-pr.yaml (hunk 2 above)
Comment (zromano) [reply .github/workflows/test-pr.yaml:20]: Oh that makes sense, smart! 👍

**Review by zromano — APPROVED** [review]

**Pull request merged**